### DISCUSSION TITLE
Colorize Brightscript within CDATA tags?

### DISCUSSION THREAD

**Discussion by joelb-viewlift** [rokucommunity/brighterscript#830] [Ideas]
Not sure if this is a limitation of XML parsers in general as none of the apps like Eclipse, Atom, VSCode have ever supported this, but it would be really nice if Brightscript within the CDATA tags could be both colorized and treated just like Brightscript within a .brs file for example, the code autocomplete does not work within the CDATA tags for Brightscript.

**Comment by TwitchBronBron** [comment]
I actually started work on this a while back, and it's functional! The problem is, we have to register these files as "scenegraph" files, which would potentially block any other plugins that depend on the file being "xml" in order to work (formatters, completions, etc...). 

https://github.com/rokucommunity/vscode-brightscript-language/pull/316

I just made this branch up to date, so you can test it with this vsix: https://github.com/rokucommunity/vscode-brightscript-language/pull/316#issuecomment-1615023576
